src/mysky/encrypted_files.test.ts

```diff
@@ -0,0 +1,210 @@
+
+    expect(directorySeed).toEqual("fa91607af922c9e57d794b7980e550fb15db99e62960fb0908b0f5af10afaf16");
+
+    expect(fileSeed).not.toEqual(directorySeed);
+  });
+});
+
+describe("deriveEncryptedFileTweak", () => {
+  it("Should derive the correct encrypted file tweak", () => {
+    // Hard-code expected value to catch breaking changes.
+    const seed = "test.hns/foo";
+    const expectedTweak = "352140f347807438f8f74edf3e0750a408f39b9f2ae4147eb9055d396b467fc8";
+
+    const result = deriveEncryptedFileTweak(seed);
+
+    expect(result).toEqual(expectedTweak);
+  });
+});
+
+const encryptedTestFilePath = "test_data/encrypted-json-file";
+const json = { message: "text" };
+const v = ENCRYPTED_JSON_RESPONSE_VERSION;
+const fullData = { _data: json, _v: v };
+const key = new Uint8Array(ENCRYPTION_KEY_LENGTH);
+const fileData = new Uint8Array(readFileSync(encryptedTestFilePath));
+
+describe("decryptJSONFile", () => {
+  it("Should decrypt the given test data", () => {
+    expect(fileData.length).toEqual(4096);
+
+    const result = decryptJSONFile(fileData, key);
+
+    expect(result).toEqual(fullData);
+  });
+
+  it("Should fail to decrypt bad data", () => {
+    expect(() => decryptJSONFile(new Uint8Array(4096), key)).toThrowError(
+      "Received unrecognized JSON response version '0', expected '1'"
+    );
+  });
+
+  it("Should fail to decrypt data with a corrupted nonce", () => {
+    const data = fileData.slice();
+    data[0]++;
+    expect(() => decryptJSONFile(data, key)).toThrowError("Could not decrypt given encrypted JSON file");
+  });
+
+  it("Should fail to decrypt data with a corrupted metadata", () => {
+    const data = fileData.slice();
+    data[ENCRYPTION_NONCE_LENGTH]++;
+    expect(() => decryptJSONFile(data, key)).toThrowError(
+      "Received unrecognized JSON response version '2', expected '1'"
+    );
+  });
+
+  it("Should fail to decrypt data with corrupted encrypted bytes", () => {
+    const data = fileData.slice();
+    data[ENCRYPTION_NONCE_LENGTH + ENCRYPTION_HIDDEN_FIELD_METADATA_LENGTH]++;
+    expect(() => decryptJSONFile(data, key)).toThrowError("Could not decrypt given encrypted JSON file");
+  });
+
+  it("Should fail to decrypt data that was not padded correctly", () => {
+    const data = fileData.slice(0, fileData.length - 1);
+    expect(data.length).toEqual(4095);
+    expect(() => decryptJSONFile(data, key)).toThrowError(
+      "Expected parameter 'data' to be padded encrypted data, length was '4095', was type 'object', value '174,167,134,21,46,207,180,245,44,139,11,69,252,151,172,83,91,4,3,35,8,124,58,113,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,216,246,27,114,110,131,41,62,175,48,210,216,97,136,48,8,79,76,33,85,100,240,252,91,188,153,39,76,217,94,167,68,133,137,216,208,244,253,19,105,85,177,254,135,12,20,24,85,185,165,5,14,89,243,15,157,237,128,66,76,41,181,192,187,199,218,199,82,43,134,154,161,91,215,191,119,33,42,7,137,188,71,228,251,245,222,30,193,...'"
```

Comment:
There's nothing a user could do about this as this would indicate a logic error on the part of the devs. Even if a user tries to decrypt non-encrypted data he should not see this.

Reply:
By "user" I meant sdk user which is a dev - even for dev this could be more readable

Reply:
SDK user shouldn't see this, either. If you have any suggestions though feel free to provide one. Ivo already gave an alternate message and this was changed in one of the latest commits.